cosmetics-web/app/models/nano_element.rb

```diff
@@ -79,4 +80,13 @@ def usage_confirmed_required?
   def notification
     nano_material.notification
   end
+
+  def unique_name_per_nanomaterial
+    nano_elements = nano_material.notification.nano_materials.map(&:nano_elements).flatten
+    other_names = nano_elements.reject { |nano_element| nano_element.id == id }.map(&:inci_name)
+    other_names = other_names.compact.map(&:downcase).map(&:strip)
+    if other_names.include? inci_name.downcase.strip
```

Comment:
I think we could turn all this into a single search in DB and let PostgreSQL deal with the matching without having to load all the notification nano elements in the application memory:

```suggestion
    nano_elements_with_same_name = NanoElement.where(nano_material: nano_material.notification.nano_materials)
                                              .where.not(id: id)
                                              .where("trim(lower(inci_name)) = ?", inci_name.downcase.strip)
    if nano_elements_with_same_name.any?
```

An example of this query running on my local machine and matching ` Nanom2 ` from DB with `nanom2` given string:
![Screenshot from 2022-04-27 18-15-44](https://user-images.githubusercontent.com/1227578/165582930-f23a9193-d689-403b-8c9d-64462069d94f.png)

Reply:
Query without spaces/formatting:
```ruby
NanoElement.where(nano_material: nano_material.notification.nano_materials).where.not(id: id).where("trim(lower(inci_name)) = ?", inci_name.downcase.strip)
```

Reply:
Its a bit more confusing - you need to go back to notification, as new relation is really one to one for `NanoElement` and `NanoMaterial`. I used ruby as there wont be notifications with many nanomaterials so the code wont be a bottleneck.